libs/sensors/include/ProximitySwitch.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "Sensor.h"
+#include "mbed.h"
+
+namespace Sensor {
+class ProximitySwitch final : Sensor {
+ public:
+  ProximitySwitch(PinName proximity_in);
+
+  // read returns 1 for HIGH, and 0 for LOW
+  float read() override{};
+  float alternateRead() override{};
+  bool getStatus() const override{};
+  [[nodiscard]] bool reset() override;
+  [[nodiscard]] bool update() override;
```

Comment:
If you are not going to use these functions but override them, then you will need to use `{}` to define an "empty" function. If you don't do this, the compiler will consider this class to be abstract because these functions were defined as pure virtual functions in the `sensor` base class.